### DISCUSSION TITLE
Migrating towards a server and agent based model

### DISCUSSION THREAD

**Discussion by drtech981** [seriousm4x/UpSnap#51] [Feature requests]
I have many sites where I have individually hosted UpSnap on a VM and cloudflare-tunneled them to the public internet (CG-NAT issues).

It would be a great idea if there was a central server, and multiple locally hosted 'agents' on the LAN itself that would fire the WOL packet through a central web-based UI.

**Comment by seriousm4x** [comment]
If I understand you correctly, you mean something like that, right?

                    +----------------+          +----------------+
                    |    Frontend    |  ----->  |   Main Server  |
                    +----------------+          +----------------+
                                                    |       |
                                                    |       |
                               Subnet 1             |       |             Subnet 2
                            +------------+----------+       +----------+------------+
                            |  Agent 1   |                             |  Agent 2   |
                        PC--+------------+--PC                     PC--+------------+--PC
                            |     |      |                             |   |   |   |
                            PC    PC     PC                            PC  PC  PC  PC

That would certainly be helpful for more complex networks, but that's currently not the goal of the project. I just want to provide a very simple way for selfhosters and their home network to wake up their devices. What you describe already takes on corporate structures, and that's currently outside my scope.

**Reply by drtech981** [reply]
Yeah, that's the model I was hoping for. 
Thanks for the heads up though !!
